Solutions/Apache Log4j Vulnerability Detection/Analytic Rules/UserAgentSearch_log4j.yaml

```diff
@@ -59,7 +59,7 @@ query: |
   (AzureDiagnostics
   | where Category in ("FrontdoorWebApplicationFirewallLog", "FrontdoorAccessLog", "ApplicationGatewayFirewallLog", "ApplicationGatewayAccessLog")
   | where userAgent_s has_any (UserAgentString) or userAgent_s matches regex UARegex
-  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by UserAgent = userAgent_s, SourceIP = clientIP_s, Type, host_s, Url = requestUri_s, httpStatus_d, ruleName_s, transactionId_g, ResourceType, ResourceId
+  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by UserAgent = userAgent_s, SourceIP = coalesce(clientIP_s,clientIp_s), Type, host_s, Url = requestUri_s, HttpStatus = coalesce(tostring(httpStatus_d),tostring(httpStatusDetails_s)), ruleName_s, transactionId_g, ResourceType, ResourceId
```

Comment:
but some customers don't have clientIP_s, however that was the only option for SourceIP. Even if you don't have the clientIp_s field, it will pick clientIP_s, that's the whole purpose. I ran the query where there's no clientIp_s and it works just fine.

Reply:
No, the query will not work. Please try this query on a workspace where the schema of AzureDiagnostics does not contain "clientIp_s" and it will fail. The requirement is legal but we would have to use column_ifexists for clientIp_s to ensure it doesn't fail for anyone.

Reply:
ok done